### PR TITLE
Hibernate5 multitenancy

### PR DESCRIPTION
**Pull request by joshiparthin**
Changes for two separate strategies

### CODE REVIEW

File: hibernate5/src/main/java/com/baeldung/hibernate/DatabaseMultitenantUtil.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.baeldung.hibernate;
+
+public class DatabaseMultitenantUtil extends HibernateMultiTenantUtil{
+    
+    
+    @Override
+    public String tenantUrl(String originalUrl, String tenant, String tenantStrategy) throws UnsupportedTenancyException {
```
Comment (KevinGilmore) [inline hibernate5/src/main/java/com/baeldung/hibernate/DatabaseMultitenantUtil.java]: The third parameter does not appear to be necessary in either of the implementations.

File: hibernate5/src/main/java/com/baeldung/hibernate/HibernateMultiTenantUtil.java [hunk 2]
```diff
@@ -97,19 +97,21 @@ private static Properties getProperties() throws IOException {
...
-            .equals("SCHEMA")) {
-            return originalUrl + String.format(SCHEMA_TOKEN, tenant);
-        } else {
-            throw new UnsupportedTenancyException("Not yet supported");
-        }
-    }
-
-    public static final String SCHEMA_TOKEN = ";INIT=CREATE SCHEMA IF NOT EXISTS %1$s\\;SET SCHEMA %1$s";
-    public static final String DEFAULT_DB_NAME = "mydb1";
+//    private static Object tenantUrl(String originalUrl, String tenant, String tenantStrategy) throws UnsupportedTenancyException {
```
Comment (KevinGilmore) [inline hibernate5/src/main/java/com/baeldung/hibernate/HibernateMultiTenantUtil.java]: Let's remove all the commented-out code.

File: hibernate5/src/main/java/com/baeldung/hibernate/SchemaMultitenantUtil.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.baeldung.hibernate;
+
+public class SchemaMultitenantUtil extends HibernateMultiTenantUtil{
+    @Override
+    public String tenantUrl(String originalUrl, String tenant, String tenantStrategy) throws UnsupportedTenancyException {
```
Comment (KevinGilmore) [inline hibernate5/src/main/java/com/baeldung/hibernate/SchemaMultitenantUtil.java]: Did you intentionally leave some of this code commented out? Need a test class for the schema strategy that exercises this code.

File: hibernate5/src/main/java/com/baeldung/hibernate/dao/SupplierDao.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        Session session = sessionFactory.withOptions().tenantIdentifier(tenant).openSession();
+        Transaction transaction = session.getTransaction();
+        
+        transaction.begin();
+        session.createSQLQuery("DROP ALL OBJECTS").executeUpdate();
+        session
+            .createSQLQuery(
+                "create table Supplier (id integer generated by default as identity, country varchar(255), name varchar(255), primary key (id))")
+            .executeUpdate();
+        Supplier genertedSupplier = generateEntityForTenant(tenant);
```
Comment (KevinGilmore) [inline hibernate5/src/main/java/com/baeldung/hibernate/dao/SupplierDao.java]: Can this (generating a test entity and persisting it) be moved outside of the DAO class? It does not seem appropriate to have this here. Maybe it belongs in a test class?

File: hibernate5/src/main/java/com/baeldung/hibernate/ConfigurableMultiTenantConnectionProvider.java [hunk 5]
```diff
@@ -34,7 +35,12 @@ protected ConnectionProvider selectConnectionProvider(String tenantIdentifier) {
     public Connection getConnection(String tenantIdentifier) throws SQLException {
         Connection connection = super.getConnection(tenantIdentifier);
         // uncomment to see option 2 for SCHEMA strategy.
-        //connection.createStatement().execute("SET SCHEMA '" + tenantIdentifier + "'");
+        if (strategy == "SCHEMA"){
```
[the remark this answers is not in the document]
Comment (joshiparthin) [reply hibernate5/src/main/java/com/baeldung/hibernate/ConfigurableMultiTenantConnectionProvider.java]: Done

File: hibernate5/src/main/java/com/baeldung/hibernate/DatabaseMultitenantUtil.java (hunk 1 above)
Comment (joshiparthin) [reply hibernate5/src/main/java/com/baeldung/hibernate/DatabaseMultitenantUtil.java]: Done. Sorry i Missed that.

File: hibernate5/src/main/java/com/baeldung/hibernate/HibernateMultiTenantUtil.java (hunk 2 above)
Comment (joshiparthin) [reply hibernate5/src/main/java/com/baeldung/hibernate/HibernateMultiTenantUtil.java]: Ok

File: hibernate5/src/main/java/com/baeldung/hibernate/SchemaMultitenantUtil.java (hunk 3 above)
Comment (joshiparthin) [reply hibernate5/src/main/java/com/baeldung/hibernate/SchemaMultitenantUtil.java]: OK. Made two Seperate Classes.

File: hibernate5/src/main/java/com/baeldung/hibernate/dao/SupplierDao.java (hunk 4 above)
Comment (joshiparthin) [reply hibernate5/src/main/java/com/baeldung/hibernate/dao/SupplierDao.java]: I remember some comment from Eugen earlier to not put the data generation code in test class. So the closest one I could find (for this tutorial code) was this place. Kindly suggest where should I put this if not here...

**Pull request closed without merging**